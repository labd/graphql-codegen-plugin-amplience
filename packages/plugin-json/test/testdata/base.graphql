type Base @amplienceContentType {
  text: String!
    @amplienceText(
      minLength: 2
      maxLength: 4
      format: markdown
      pattern: "/[\\w\\-]+"
      examples: ["one example", "another example"]
    )
  optionalText: String
  textList: [String!]! @amplienceList(minItems: 1, maxItems: 10)
  integer: Int! @amplienceNumber(minimum: 0, maximum: 10)
  optionalInteger: Int
  float: Float! @amplienceNumber(minimum: 0, maximum: 10)
  optionalFloat: Float
  boolean: Boolean!
  optionalBoolean: Boolean
  constString: String @amplienceConst(item: "const")
  constArray: [String!]! @amplienceConst(items: ["this", "is", "const"])
  image: AmplienceImage!
  video: AmplienceVideo!
  enum: Enum
  a: A
  bs: [B]
  inlined: Inlined
  union: Union
  linkedA: A @amplienceLink
  referencedB: B @amplienceReference
}

enum Enum {
  a
  b
}

union Union = A | B

type A @amplienceContentType {
  a: String!
}

type B @amplienceContentType {
  b: String!
}

type Inlined {
  inlined: String!
}

type Localized @amplienceContentType {
  string: String @amplienceLocalized
  boolean: Boolean @amplienceLocalized
  int: Int @amplienceLocalized
  float: Float @amplienceLocalized
  image: AmplienceImage @amplienceLocalized
  video: AmplienceVideo @amplienceLocalized
}
