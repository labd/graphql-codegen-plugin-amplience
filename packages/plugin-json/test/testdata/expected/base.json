{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "https://schema-examples.com/base",
  "title": "Base",
  "description": "Base",
  "allOf": [
    {
      "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content"
    }
  ],
  "type": "object",
  "properties": {
    "_meta": {
      "type": "object",
      "title": "Delivery Key",
      "properties": {
        "deliveryKey": {
          "type": "string",
          "title": "Delivery Key",
          "description": "Set a delivery key for this content item"
        }
      }
    },
    "text": {
      "title": "Text",
      "type": "string",
      "format": "markdown",
      "minLength": 2,
      "maxLength": 4,
      "pattern": "/[\\w\\-]+",
      "examples": ["one example", "another example"]
    },
    "optionalText": {
      "title": "Optional Text",
      "type": "string"
    },
    "textList": {
      "title": "Text List",
      "minItems": 1,
      "maxItems": 10,
      "type": "array",
      "items": {
        "type": "string"
      }
    },
    "boolean": {
      "title": "Boolean",
      "type": "boolean"
    },
    "optionalBoolean": {
      "title": "Optional Boolean",
      "type": "boolean"
    },
    "integer": {
      "maximum": 10,
      "minimum": 0,
      "title": "Integer",
      "type": "integer"
    },
    "optionalInteger": {
      "title": "Optional Integer",
      "type": "integer"
    },
    "float": {
      "maximum": 10,
      "minimum": 0,
      "title": "Float",
      "type": "number"
    },
    "optionalFloat": {
      "title": "Optional Float",
      "type": "number"
    },
    "constString": {
      "title": "Const String",
      "type": "string",
      "const": "const"
    },
    "constArray": {
      "title": "Const Array",
      "type": "array",
      "items": {
        "type": "string"
      },
      "const": ["this", "is", "const"]
    },
    "enum": {
      "title": "Enum",
      "type": "string",
      "enum": ["a", "b"]
    },
    "a": {
      "title": "A",
      "type": "object",
      "allOf": [
        {
          "$ref": "https://schema-examples.com/a"
        }
      ]
    },
    "bs": {
      "title": "Bs",
      "type": "array",
      "items": {
        "type": "object",
        "allOf": [
          {
            "$ref": "https://schema-examples.com/b"
          }
        ]
      }
    },
    "inlined": {
      "title": "Inlined",
      "type": "object",
      "properties": {
        "inlined": {
          "title": "Inlined",
          "type": "string"
        }
      },
      "propertyOrder": ["inlined"],
      "required": ["inlined"]
    },
    "image": {
      "title": "Image",
      "allOf": [
        {
          "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link"
        }
      ]
    },
    "video": {
      "title": "Video",
      "allOf": [
        {
          "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/video-link"
        }
      ]
    },
    "linkedA": {
      "title": "Linked A",
      "allOf": [
        {
          "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content-link"
        },
        {
          "properties": {
            "contentType": {
              "enum": ["https://schema-examples.com/a"]
            }
          }
        }
      ]
    },
    "referencedB": {
      "allOf": [
        {
          "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content-reference"
        },
        {
          "properties": {
            "contentType": {
              "enum": ["https://schema-examples.com/b"]
            }
          }
        }
      ],
      "title": "Referenced B"
    },
    "union": {
      "title": "Union",
      "allOf": [
        {
          "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content-link"
        },
        {
          "properties": {
            "contentType": {
              "enum": [
                "https://schema-examples.com/a",
                "https://schema-examples.com/b"
              ]
            }
          }
        }
      ]
    }
  },
  "required": [
    "text",
    "textList",
    "integer",
    "float",
    "boolean",
    "constArray",
    "image",
    "video"
  ],
  "propertyOrder": [
    "_meta",
    "text",
    "optionalText",
    "textList",
    "integer",
    "optionalInteger",
    "float",
    "optionalFloat",
    "boolean",
    "optionalBoolean",
    "constString",
    "constArray",
    "image",
    "video",
    "enum",
    "a",
    "bs",
    "inlined",
    "union",
    "linkedA",
    "referencedB"
  ]
}
